src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseSimulationResource.java

```diff
@@ -0,0 +1,190 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import static de.tum.in.www1.artemis.config.Constants.SHORT_NAME_PATTERN;
+import static de.tum.in.www1.artemis.config.Constants.TITLE_NAME_PATTERN;
+import static de.tum.in.www1.artemis.web.rest.util.ResponseUtil.forbidden;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.CourseService;
+import de.tum.in.www1.artemis.service.ProgrammingExerciseSimulationService;
+import de.tum.in.www1.artemis.service.UserService;
+import de.tum.in.www1.artemis.service.feature.Feature;
+import de.tum.in.www1.artemis.service.feature.FeatureToggle;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+
+@RestController
+@RequestMapping(ProgrammingExerciseSimulationResource.Endpoints.ROOT)
+/**
+ * Only for local development
+ * Simulates the creation of a programming exercise without a local setup
+ */
+public class ProgrammingExerciseSimulationResource {
+
+    private final Logger log = LoggerFactory.getLogger(ProgrammingExerciseResource.class);
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private static final String ENTITY_NAME = "programmingExercise";
+
+    private final CourseService courseService;
+
+    private final UserService userService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final String packageNameRegex = "^[a-z][a-z0-9_]*(\\.[a-z0-9_]+)+[0-9a-z_]$";
```

Comment:
This already exists in the `ProgrammingExerciseResource`, maybe we create a Constant to avoid duplication.
See `TITLE_NAME_PATTERN` for an example.